lifemonitor/api.yaml

```diff
@@ -183,10 +431,150 @@ components:
         - version
         - roc_link
 
+    WorkflowStatus:
+      type: object
+      properties:
+        workflow:
+          schema:
+            type: object
+            properties:
+              uuid:
+                type: string
+              version:
+                type: string
+              name:
+                type: string
+            required:
+              - uuid
+              - version
+        aggregate_test_status:
+          $ref: '#/components/schemas/AggregateTestStatus'
+        latest_builds:
+          description: "Latest builds, one for each test instance"
+          type: array
+          items:
+            $ref: '#/components/schemas/BuildSummary'
+
+    AggregateTestStatus:
+      type: string
+      enum:
+        - "all-passing"
+        - "some-passing"
+        - "all-failing"
+        - "not-available"
+
     TestSuite:
       type: object
       properties:
+        uuid:
+          type: string
         test_suite_metadata:
           type: object
+          description: "To be determined."
+        instances:
+          type: array
+          items:
+            $ref: '#/components/schemas/SuiteInstance'
       required:
+        - uuid
         - test_suite_metadata
+        - instances
+
+    TestSuiteStatus:
+      type: object
+      properties:
+        suite_uuid:
+          type: string
+        status:
+          $ref: '#/components/schemas/AggregateTestStatus'
+        latest-builds:
+          type: array
+          items:
+            $ref: '#/components/schemas/BuildSummary'
+      required:
+        - suite_uuid
+        - status
+
+    BuildStatus:
+      type: string
+      enum:
+        - passed
+        - failed
+        - error
+        - running
+        - waiting
+
+    BuildSummary:
+      type: object
+      properties:
+        build_id:
+          type: string
+        suite_uuid:
+          type: string
+        status:
+          $ref: '#/components/schemas/BuildStatus'
+        instance:
+          $ref: '#/components/schemas/SuiteInstance'
+        timestamp:
+          $ref: '#components/schema/Timestamp'
+        lastlogs:
+          type: string
+          description: "Last few lines of build log, if available"
+      required:
+        - build_id
+        - suite_uuid
+        - instance
+        - timestamp
+
+    Build:
+      type: object
+      properties:
+        build_id:
+          type: string
+        suite_uuid:
+          type: string
+        status:
+          $ref: '#/components/schemas/BuildStatus'
+        instance:
+          $ref: '#/components/schemas/SuiteInstance'
+        timestamp:
+          $ref: '#components/schema/Timestamp'
+          # other things to be added
+      required:
+        - build_id
+        - suite_uuid
+        - status
+        - instance
+        - timestamp
+
+
+    SuiteInstance:
```

Comment:
In the case of internal testing, how do we use this to model the internal instance? By using special values for the properties?

Reply:
Yes, and I don't think I'd even call them special.  I think tests running on LM-managed or external infrastructure should be treated the same.  Rather, we add some additional metadata to express what LM can do with an given instance -- e.g., is it authorized to trigger a new build?